dotcom-rendering/src/web/components/Liveness.importable.tsx

```diff
@@ -0,0 +1,186 @@
+	// TODO Call existing hydration solution targetted to these elements
+	// Maybe we can make doHydration work for all gu-islands things
+	// where the gu-hydrated data attribute isn't set?
+}
+
+/**
+ * insertNewBlocks takes html and inserts it at the top of the liveblog
+ *
+ * @param {string} html The block html to be inserted
+ * @returns void
+ */
+function insertNewBlocks(html: string) {
+	const latestBlock = document.querySelector('#maincontent :first-child');
+	if (!latestBlock) return;
+	latestBlock.insertAdjacentHTML('beforebegin', html);
+}
+
+/**
+ * revealNewBlocks - style any blocks that have been inserted but are hidden such that
+ * they are revealed
+ */
+function revealNewBlocks() {
+	console.log('revealNewBlocks');
+}
+
+/**
+ * This abstracts the code to construct the url we use for polling. Of note is the fact
+ * we have put `latestBlockId` in react state; by doing this we are able to change the url
+ * that SWR uses for the request whilst still benefitting from the polling logic
+ *
+ * @returns string
+ */
+function getKey(
+	pageId: string,
+	ajaxUrl: string,
+	filterKeyEvents: boolean,
+	latestBlockId: string,
+): string {
+	// Construct the url to poll
+	const url = new URL(`${pageId}.json`, ajaxUrl);
+	url.searchParams.set('lastUpdate', latestBlockId);
+	url.searchParams.set('isLivePage', 'true');
+	url.searchParams.set('filterKeyEvents', filterKeyEvents ? 'true' : 'false');
+	return url.href;
+}
+
+const topOfBlog: Element | null | false =
+	!isServer && window.document.querySelector('[data-gu-marker=top-of-blog]');
+
+/**
+ * This allows us to make decisions in javascript based on if the reader
+ * has the top of the blog in view or not
+ *
+ * @returns boolean
+ */
+function topOfBlogVisible(): boolean {
+	return !!topOfBlog && !!topOfBlog.classList.contains('in-viewport');
+}
+
+export const Liveness = ({ pageId, webTitle, ajaxUrl }: Props) => {
+	const [showToast, setShowToast] = useState(false);
+	const [noOfNewPosts, setNoOfNewPosts] = useState(0);
+	const [latestBlockId, setLatestBlockId] = useState(
+		// By default we use the first (latest) block id on the page
+		document.querySelector('#maincontent :first-child')?.id || '',
+	);
+
+	// Read current url to get filterKeyEvents
+	const params = new URLSearchParams(window.location.search);
+	const filterKeyEvents = !!params.get('filterKeyEvents');
+
+	// useApi returns { data, loading, error } but we're not using them here
+	useApi<{
+		numNewBlocks: number;
+		html: string;
+	}>(getKey(pageId, ajaxUrl, filterKeyEvents, latestBlockId), {
+		refreshInterval: 15000,
+		refreshWhenHidden: true,
+		// onSuccess runs (once) after every successful api call. This is useful because it
+		// allows us to avoid the problems of imperative code being executed multiple times
+		// inside react's declarative structure (things get re-rendered when any state changes)
+		onSuccess: (data: {
+			numNewBlocks: number;
+			html: string;
+			mostRecentBlockId: string;
```

Comment:
This may be an issue with the type definition of `useApi` rather than anything to do with this PR, but why is this object different to the type parameter passed to `useApi`? It's got an additional `mostRecentBlockId` field 🤔 

Should `SWRConfiguration` in the `useApi.tsx` module take `T` as a type parameter?

Reply:
Good spot. They're actually typing two different things. We actually don't need that type which is being passed to `useApi` because we're not using it (we don't reference the return values from `useApi` anywhere)

Reply:
I made a change to remove the unused type